pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -1209,52 +1206,80 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
+			},
 		}
-		server := httptest.NewServer(&handler)
-		defer server.Close()
-		client := clientset.NewForConfigOrDie(&restclient.Config{Host: server.URL, ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
 		informerFactory := informers.NewSharedInformerFactory(client, 0)
 
-		if _, err := factory.NewConfigFactory(&factory.ConfigFactoryArgs{
-			Client:                         client,
-			NodeInformer:                   informerFactory.Core().V1().Nodes(),
-			PodInformer:                    informerFactory.Core().V1().Pods(),
-			PvInformer:                     informerFactory.Core().V1().PersistentVolumes(),
-			PvcInformer:                    informerFactory.Core().V1().PersistentVolumeClaims(),
-			ReplicationControllerInformer:  informerFactory.Core().V1().ReplicationControllers(),
-			ReplicaSetInformer:             informerFactory.Apps().V1().ReplicaSets(),
-			StatefulSetInformer:            informerFactory.Apps().V1().StatefulSets(),
-			ServiceInformer:                informerFactory.Core().V1().Services(),
-			PdbInformer:                    informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
-			StorageClassInformer:           informerFactory.Storage().V1().StorageClasses(),
-			CSINodeInformer:                informerFactory.Storage().V1beta1().CSINodes(),
-			HardPodAffinitySymmetricWeight: v1.DefaultHardPodAffinitySymmetricWeight,
-			DisablePreemption:              false,
-			PercentageOfNodesToScore:       schedulerapi.DefaultPercentageOfNodesToScore,
-		}).CreateFromConfig(policy); err != nil {
+		sched, err := scheduler.New(
+			client,
+			informerFactory.Core().V1().Nodes(),
+			informerFactory.Core().V1().Pods(),
+			informerFactory.Core().V1().PersistentVolumes(),
+			informerFactory.Core().V1().PersistentVolumeClaims(),
+			informerFactory.Core().V1().ReplicationControllers(),
+			informerFactory.Apps().V1().ReplicaSets(),
+			informerFactory.Apps().V1().StatefulSets(),
+			informerFactory.Core().V1().Services(),
+			informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
+			informerFactory.Storage().V1().StorageClasses(),
+			informerFactory.Storage().V1beta1().CSINodes(),
+			nil,
+			algorithmSrc,
+			make(chan struct{}),
+			schedulerframework.NewDefaultRegistry(),
+			nil,
+			[]kubeschedulerconfig.PluginConfig{},
+		)
+		if err != nil {
 			t.Errorf("%s: Error constructing: %v", v, err)
 			continue
 		}
-	}
-
-	if !seenPredicates.HasAll(registeredPredicates.List()...) {
-		t.Errorf("Registered predicates are missing from compatibility test (add to test stanza for version currently in development): %#v", registeredPredicates.Difference(seenPredicates).List())
-	}
-	if !seenPriorities.HasAll(registeredPriorities.List()...) {
-		t.Errorf("Registered priorities are missing from compatibility test (add to test stanza for version currently in development): %#v", registeredPriorities.Difference(seenPriorities).List())
+		schedPredicates := sets.NewString()
```

Comment:
Not sure if that is the correct way to compare predicates and prioritizers. Would like to hear others recommendations for this.

Reply:
I would just get rid of `expectedPolicy` altogether, and have `wantPredicates: sets.NewString(...)`. Same for prioritizers. The way you are obtaining the names from the Predicates map LGTM.

Reply:
Can we get rid of expectedPolicy but keep wantPredicates as []schedulerapi.PredicatePolicy to avoid having a huge number of changes in the PR which do not seem necessary to me? or is it ok to have such large number of changes?

Reply:
If you want you can make a parent PR to change just that, but I think it's ok to have it in this PR.
Just update the description indicating that we are fixing the test to check for the fields in the instantiated scheduler instead of the decoding.